### ISSUE TITLE
KMS Encryption for kubernetes secrets does not work when defined in cluster.yaml

### ISSUE THREAD

**Issue by practiv-jfox** [weaveworks/eksctl#4504]
**What were you trying to accomplish?**

Wanted to create a new EKS cluster with KMS encryption enabled 

**What happened?**

These was no errors reported in the EKS cluster creation.
Encryption was not enabled

**How to reproduce it?**

Using EKSCTL 0.72.0 and the cluster.yaml method as per the instructions here. https://eksctl.io/usage/kms-encryption/
My cluster config matches the secretsEncryption section of documentation exactly. 

**Logs**

Proof of no encrypted secrets. 

```

$ kubectl get nodes
NAME                                              STATUS   ROLES    AGE   VERSION
ip-10-0-105-170.ap-southeast-2.compute.internal   Ready    <none>   66m   v1.20.11-eks-f17b81
ip-10-0-115-217.ap-southeast-2.compute.internal   Ready    <none>   66m   v1.20.11-eks-f17b81
ip-10-0-125-173.ap-southeast-2.compute.internal   Ready    <none>   66m   v1.20.11-eks-f17b81

$ kubectl --all-namespaces=true get secrets -o yaml | grep eksctl.io/kms-encryption-timestamp -c
0 
```

**Anything else we need to know?**

OS: Amazon Linux 2
Authentication: IAM user with AWS access and secret key with recommended IAM permissions as per the documentation.
Download binary directly from github.com

encrypt does work via command line so there is no issues with my KMS key.

it does work if i use command line method `$ eksctl utils enable-secrets-encryption --cluster=kms-cluster --key-arn=arn:aws:kms:us-west-2:<account>:key/<key> --region=<region>`

**Versions**

```
eksctl version 0.72.0
```

**Comment by cPu1** [comment]
>Proof of no encrypted secrets.

>kubectl --all-namespaces=true get secrets -o yaml | grep eksctl.io/kms-encryption-timestamp -c
0 

The annotation `eksctl.io/kms-encryption-timestamp ` is applied to trigger encryption of existing secrets as part of the `eksctl utils enable-secrets-encryption` command, not when running `eksctl create cluster`. In your first example, you should check if secrets encryption is enabled by running `aws eks describe-cluster`.

**Comment by cPu1** [comment]
I'm removing the bug label for now but if you're able to confirm that the cluster ends up without secrets encryption enabled by doing a `DescribeCluster` call, we can re-add it.

**Comment by practiv-jfox** [comment]
Hi Chetan

So running `aws eks describe-cluster` does show that encryptionConfig is configured with the key ive provided in the cluster.yaml. 

Can you please include this test in the documentation as i cant be the only one to raise this concern. 

I will now close off this issue.

Thanks Jared

**Comment by rajnikantMtech** [comment]
In my case, I have created EKS cluster without Secret encryption enabled. Then I have enabled it using key ARN using below command
eksctl utils enable-secrets-encryption --cluster eks-kms-test-cluster --key-arn arn:aws:kms:region:-<acc. number>:key/<key-id>

It runs forever even though I have very few default secrets on newly created cluster. Means It logs cloudtrail "Encrypt" event continuously. Don't know why.

Do anyone have any idea ?
